### PR TITLE
implemented user story that allows user to see the listings && pagination included for listings issue63

### PR DESCRIPTION
**Pull request by Mik3Qiao**
implemented user story that allows user to see the listings && pagination included for listings

### CODE REVIEW

File: frontend/package.json [hunk 1]
```diff
@@ -5,21 +5,25 @@
...
+    "moment": "^2.29.1",
     "react": "^17.0.1",
     "react-bootstrap": "^1.4.3",
     "react-dom": "^17.0.1",
     "react-redux": "^7.2.2",
     "react-router-dom": "^5.2.0",
     "react-scripts": "4.0.1",
     "redux": "^4.0.5",
-    "web-vitals": "^0.2.4"
+    "web-vitals": "^0.2.4",
```
Comment (NathanCarriere) [inline frontend/package.json:25]: Question: Are these added automatically whenever you download something with npm?

File: frontend/src/components/Listing/listing.jsx [hunk 2]
```diff
@@ -2,62 +2,273 @@ import React, { Component } from "react";
...
+
+import "../../styles/Listing.css";
 
 class Listing extends Component {
-  
+
   componentDidMount (){
-    this.props.setBedRoom(2);
-    this.props.setWashRoom(10);
+    // axios.get("http://localhost:3005/apartment1")
```
Comment (NathanCarriere) [inline frontend/src/components/Listing/listing.jsx:27]: Usually not good practice to commit commented-out code.

File: frontend/src/components/Listing/listing.jsx [hunk 3]
```diff
@@ -2,62 +2,273 @@ import React, { Component } from "react";
...
+    //   this.setState({
+    //     imageSource: response.data[0].source
+    //   })
+    // })
+
+    // this.setState({
+    //   imageSource: this.getEmergencyFoundImg(apartment1)
+    // })
+    axios.get("http://localhost:3005/listingArray")
+    .then(response =>{
```
Comment (NathanCarriere) [inline frontend/src/components/Listing/listing.jsx:38]: .then should probably be indented.

File: frontend/src/components/Listing/listing.jsx [hunk 4]
```diff
@@ -2,62 +2,273 @@ import React, { Component } from "react";
...
-          src = {apartment1}
-          alt="apartment 1"
-        />
+      <div className = "listingContent">
+        <div className="leftPanel">
+          {/* {this.props.listingArray.map(listingDetail, index) => (
+
+          )} */}
+          {this.props.listingArray && this.props.listingArray.length !== 0 
+          ?
```
Comment (NathanCarriere) [inline frontend/src/components/Listing/listing.jsx:72]: It's hard to follow what this ternary operator is doing

File: frontend/src/components/Listing/listing.jsx [hunk 5]
```diff
@@ -2,62 +2,273 @@ import React, { Component } from "react";
...
+          {/* {this.props.listingArray.map(listingDetail, index) => (
+
+          )} */}
+          {this.props.listingArray && this.props.listingArray.length !== 0 
+          ? 
+            this.props.listingArray
+            .slice((this.props.pageNum - 1) * 10, this.props.pageNum * 10)
+            .map((listingDetail, index)=>{
+            // return
+            const currIndex = this.props.pageNum * 10 + index;
```
Comment (NathanCarriere) [inline frontend/src/components/Listing/listing.jsx:77]: Should be indented

File: frontend/src/components/Listing/listing.jsx [hunk 6]
```diff
@@ -2,62 +2,273 @@ import React, { Component } from "react";
...
+                <div className = "listingTextAndIcon">
+                  <span className = "listingHeader">
+                  
+                    {/* This is for the listing title area */}
+                    <span className = "listingTitle">
+                      {listingDetail.title}
+                    </span>
+
+                    <Typography
+                      type="title"
```
Comment (NathanCarriere) [inline frontend/src/components/Listing/listing.jsx:113]: Some element attributes in this file have spaces around the `=` and some do not. I think it could be good to get ESLint and have it run on save since that would help make this consistent.

File: frontend/package.json (hunk 1 above)
Comment (Mik3Qiao) [reply frontend/package.json:25]: npm install will install those packages, and when u use npm install or npm i xxx, that xxx package will be added to this package.json and packagelock.json

File: frontend/src/components/Listing/listing.jsx (hunk 2 above)
Comment (Mik3Qiao) [reply frontend/src/components/Listing/listing.jsx:27]: we should, this is for post request which is for the rest API, as long as we don't have commented code in master during the end of an iteration, I think it should be fine.

File: frontend/src/components/Listing/listing.jsx [hunk 7]
```diff
@@ -2,62 +2,273 @@ import React, { Component } from "react";
...
+                    <Divider orientation="vertical" flexItem />
+
+                    <div
+                      className = "listingPrice"
+                    >
+                      ${listingDetail.price}
+                    </div>
+
+                  </span>
+
```
[the remark this answers is not in the document]
Comment (Mik3Qiao) [reply frontend/src/components/Listing/listing.jsx:178]: thought white space would make it seem less tight, or seems kind of messy. Anyway u prefer

File: frontend/src/components/Listing/listing.jsx (hunk 3 above)
Comment (Mik3Qiao) [reply frontend/src/components/Listing/listing.jsx:38]: I thought not indenting makes more sense... anyway would be fine

**Review by AzizulHakim96 — APPROVED** [review]

File: frontend/src/components/Listing/listing.jsx (hunk 4 above)
Comment (Mik3Qiao) [reply frontend/src/components/Listing/listing.jsx:72]: it is done in jsx this way, I wasn't used to this style at the first place.

File: frontend/src/components/Listing/listing.jsx (hunk 5 above)
Comment (Mik3Qiao) [reply frontend/src/components/Listing/listing.jsx:77]: ...

Comment (Mik3Qiao) [reply frontend/src/components/Listing/listing.jsx:77]: I'll correct it :D

File: frontend/src/components/Listing/listing.jsx (hunk 6 above)
Comment (Mik3Qiao) [reply frontend/src/components/Listing/listing.jsx:113]: I don't think the spaces around the attributes would make any difference

**Comment by RonnieDsouza** [comment]
Hey Mike could you make the PR Title a bit more descriptive? Instead of your name can you say in ~5 words what the PR is about?

File: frontend/src/components/Listing/listing.jsx (hunk 7 above)
[the remark this answers is not in the document]
Comment (NathanCarriere) [reply frontend/src/components/Listing/listing.jsx:180]: For other areas, I think it's fine. But between closing tags, I'm not sure if we have to worry much about clutter. It's also inconsistent with other parts of code (see lines 230-232).

**Review by NathanCarriere — APPROVED** [review]

**Review by RonnieDsouza — APPROVED** [review]
Thanks! Looks Good!
